pkg/cli/plugin_cmd.go

```diff
@@ -172,12 +233,22 @@ func getHelpArguments() []string {
 
 // setupPluginEnv prepares some extra environment variables
 // that communicate certain information to plugins.
-func setupPluginEnv() {
+func setupPluginEnv(srcHierarchy, dstHierarchy []string) {
 	env := make(map[string]string, 10)
 
 	// The location of the tanzu binary
 	env["TANZU_BIN"] = os.Args[0]
 
+	// Information about if command invocation is via a mapped command
+	numParts := len(dstHierarchy)
+	if numParts > 0 {
+		env["TANZU_CLI_INVOKED_COMMAND"] = dstHierarchy[numParts-1]
```

Comment:
I haven't tried to figure out the problem but this does not work if I'm invoking a sub-command of a mapped command.
For example, I have mapped `tanzu builder cli` to `tanzu cli`.  If I run `tanzu cli -h` things are good but not if I run `tanzu cli add-plugin -h`:
```
$ tz cli add-plugin -h
Add a plugin to a repository

Usage:
  tanzu cli
```
I tested and the variables are:
```
TANZU_CLI_INVOKED_GROUP:
TANZU_CLI_INVOKED_COMMAND: cli
TANZU_CLI_COMMAND_MAPPED_FROM: cli
```

Reply:
This is an issue with the usage generation. I will address it in a followup.